docs/articles/documentation/how-it-works/README.md

```diff
@@ -0,0 +1,154 @@
+
+    ```js
+    const getUrl = ClientFunction(() => {
+        return window.location.href;
+    });
+
+    await t.expect(getUrl).eql('https://example.com');
+    ```
+
+* *Inject scripts.* TestCafe also allows you to inject custom code into the tested page. You can use the injected scripts to add helper functions or mock browser API.
+
+    ```js
+    fixture `My Fixture`
+        .page `https://example.com`
+        .clientScripts('assets/jquery.js', 'scripts/location-mock.js');
+    ```
+
+Note that a best practice rule is to avoid page modifications in custom client-side code. Actions that change DOM should be performed through server-side API only because there's no reliable way to perform them manually without interfering with TestCafe internal mechanisms. The rule is to use client-side code only to explore the page, find, and return information to the server.
+
+## Reverse Proxy
+
+You may notice that when you run TestCafe tests, the browser's address bar shows a URL that does not belong to your website. This is because TestCafe proxies the tested pages.
+
+TestCafe reverse proxy ([testcafe-hammerhead](https://github.com/DevExpress/testcafe-hammerhead)) runs locally on your computer. It enables TestCafe to add automation scripts in a way that neither the existing page code nor the resources it communicates with can tell that the page has been modified.
+
+When TestCafe proxies the tested page, it injects automation scripts and rewrites all URLs inside (in the markup, scripts, etc.) to point to the proxy. When the browser refers to these new URLs, the original resources are also proxied and modified in the same way to let automation continue seamlessly. TestCafe also mocks the browser API to conceal automation scripts from the rest of the page code.
+
+The proxying mechanism ensures that the page appears to be hosted at the original URL even to the test code. This is why you can use your website's URLs in tests, and disregard the URL you see in the browser's address bar.
+
+## Automation Scripts
+
+TestCafe automation scripts emulate end-user actions on the tested page. They are designed to imitate the behavior of different browsers in every detail. Automation scripts dispatch events, focus elements or process attributes the way the browser that runs them would do. They abstract out browser specifics and enable you to write cross-browser high-level test code.
+
+## Isolated Test Runs
+
+TestCafe isolates each test run from subsequent tests and tests that run in parallel.
+
+After a test is completed, TestCafe resets the browser state:
+
+* deletes cookies
+* clears local and session storages (IndexedDB is preserved)
```

Comment:
The fact that 'IndexedDB is preserved' is a problem - https://github.com/DevExpress/testcafe-hammerhead/issues/1429
Someday, we will implement it.

I propose to remove this phrase because it is a problem and contradicts the paragraph idea (delete, clear something).